editors/code/src/debug.ts

```diff
@@ -6,11 +6,20 @@ import type * as ra from "./lsp_ext";
 import { Cargo } from "./toolchain";
 import type { Ctx } from "./ctx";
 import { prepareEnv } from "./run";
-import { execute, isCargoRunnableArgs, unwrapUndefinable } from "./util";
+import { execute, isCargoRunnableArgs, log, unwrapUndefinable } from "./util";
 import type { Config } from "./config";
 
 const debugOutput = vscode.window.createOutputChannel("Debug");
 
+
+// Every debug session with the same id (restart-candidate) should be in this list
+let debugSessionIdBuffer: string[] = [];
```

Comment:
Likewise this seems unnecessary, we should only need to track the last session and compare that?